src/Xabe.FFmpeg/FFmpeg.cs

```diff
@@ -40,14 +48,31 @@ private void FindAndValidateExecutables()
 
             if (!string.IsNullOrWhiteSpace(ExecutablesPath))
             {
-                FFprobePath = new DirectoryInfo(ExecutablesPath).GetFiles()
-                                                          .FirstOrDefault(x => x.Name.ToLower()
-                                                                                .Contains(_ffprobeExecutableName.ToLower()))?
-                                                          .FullName;
-                FFmpegPath = new DirectoryInfo(ExecutablesPath).GetFiles()
-                                                         .FirstOrDefault(x => x.Name.ToLower()
-                                                                               .Contains(_ffmpegExecutableName.ToLower()))?
-                                                         .FullName;
+                var files = new DirectoryInfo(ExecutablesPath).GetFiles();
+                switch (FilterMethod)
```

Comment:
Create Func<string, IFormatProvider, bool> compareMethod and set only that variable in switch. Instead of Contains/Equals/StartsWith use that function.

You use FormatProvider in wrong method. It should be in Contains/Equals/StartsWith not in IsExecutable